net/discovery.go

```diff
@@ -154,27 +156,46 @@ func (d *discovery) advertise() time.Duration {
 	return defaultAdvertiseTTL
 }
 
-func (d *discovery) discover(provides types.ProvidesCoin,
-	searchTime time.Duration) ([]peer.AddrInfo, error) {
-	log.Debugf("attempting to find DHT peers that provide [%s] for %vs...",
-		provides,
-		searchTime.Seconds(),
-	)
+func (d *discovery) discoverLoop() {
+	const discoverLoopDuration = time.Minute
+	timer := time.NewTicker(discoverLoopDuration)
+
+	for {
+		select {
+		case <-d.ctx.Done():
+			timer.Stop()
+			return
+		case <-timer.C:
+			if len(d.h.Network().Peers()) >= defaultMinPeers {
+				continue
+			}
+
+			// if our peer count is low, try to find some peers
+			findPeersTimer := time.NewTimer(discoverLoopDuration)
 
-	peerCh, err := d.rd.FindPeers(d.ctx, string(provides))
+			_, err := d.findPeers("", findPeersTimer.C)
+			if err != nil {
+				log.Errorf("failed to find peers: %s", err)
+			}
+
+			findPeersTimer.Stop()
+		}
+	}
+}
+
+func (d *discovery) findPeers(provides string, done <-chan time.Time) ([]peer.AddrInfo, error) {
```

Comment:
Just an idea to reduce code complexity:

You could pass in the timeout as a duration instead of a channel. This function could  then just wrap `d.ctx` using `context.WithTimeout` instead of creating a new channel.

Reply:
it would be nice to have the two cases be handled differently imo:
```
		select {
		case <-d.ctx.Done():
			return peers, d.ctx.Err()
		case <-done:
			return peers, nil
               // code omitted
```

what do you think?

Reply:
Sorry, I edited my comment after seeing that the 2 cases were handled differently and you had time to reply before I finished. :) See above.

Reply:
oh lol, no worries! yeah, I guess we expect the timeout to always be hit basically, so we don't want to return an error